lib/fluent/plugin/in_windows_eventlog2.rb

```diff
@@ -66,9 +67,16 @@ def initalize
     def configure(conf)
       super
       @chs = []
+      @allChs = Winevt::EventLog::Channel.new
 
+      if read_all_channels
+        @allChs.each do |allChs|
+          @chs.push(allChs)
+        end
+      end
+          
       @read_existing_events = @read_from_head || @read_existing_events
-      if @channels.empty? && @subscribe_configs.empty?
+      if @channels.empty? && @subscribe_configs.empty? && !read_all_channels
```

Comment:
ditto.

Reply:
Do you want to keep it like !@read_all_channels or skip it completely?

Reply:
Yes. Using `!@read_all_channels` in this if clause is better.